### DISCUSSION TITLE
Is streaming possible?

### DISCUSSION THREAD

**Discussion by imc-db** [apiflask/apiflask#609] [Q&A] [answered]
I use apiflask for a chatbot and one endpoint should stream the response. (OpenAI Completion)
How to activate streaming?

```
# Stream answer
data_stream = client.chat.completions.create(
    model=model_conversation, messages=data["messages"], stream=True
)
for chunk in data_stream:
    if chunk.choices[0].delta.content is not None:
        yield f"{chunk.choices[0].delta.content}"
yield "[END_OF_STREAM]"
```

**Comment by imc-db** [comment]
This example is working -> but response is not streaming. Response is visible after 5 seconds.

![image](https://github.com/user-attachments/assets/be6aa89f-c1a4-44e2-8554-e54268d78869)

**Comment by imc-db** [comment, marked as answer]
I got it -> content_type was wrong ...
![image](https://github.com/user-attachments/assets/d1e5c718-b652-42df-8fd3-2b628e6969e5)

**Comment by Bryanthelol** [comment]
helpful for me, thx.
